src/aap_eda/tasks/orchestrator.py

```diff
@@ -121,43 +135,193 @@ def _run_request(
+                )
+                manager._set_activation_status(ActivationStatus.UNKNOWN, msg)
+                LOGGER.warning(msg)
+                return
+
+        unique_enqueue(
+            queue_name,
+            job_id,
+            _manage,
+            process_parent_type,
+            process_parent_id,
+        )
+
+    @staticmethod
+    def get_most_free_queue_name() -> str:
+        """Return the queue name with the least running processes."""
+
+        if len(settings.RULEBOOK_WORKER_QUEUES) == 1:
+            return settings.RULEBOOK_WORKER_QUEUES[0]
+
+        queue_counter = Counter()
+
+        for queue_name in settings.RULEBOOK_WORKER_QUEUES:
+            if not is_rulebook_queue_healthy(queue_name):
+                continue
+            running_processes_count = models.RulebookProcess.objects.filter(
+                status=ActivationStatus.RUNNING,
+                rulebookprocessqueue__queue_name=queue_name,
+            ).count()
+            queue_counter[queue_name] = running_processes_count
+
+        if not queue_counter:
+            raise QueueNotFoundError(
+                "No healthy queue found to dispatch the request",
+            )
+
+        return queue_counter.most_common()[-1][0]
+
+    @staticmethod
+    def get_queue_name_by_parent_id(
+        process_parent_type: ProcessParentType,
+        process_parent_id: int,
+    ) -> Optional[str]:
+        """Return the queue name associated with the process ID."""
+        try:
+            parent_process = get_process_parent(
+                process_parent_type,
+                process_parent_id,
+            )
+            process = parent_process.latest_instance
+        except ObjectDoesNotExist:
+            raise ValueError(
+                f"No {process_parent_type} found with ID {process_parent_id}"
+            ) from None
+        except models.RulebookProcess.DoesNotExist:
+            raise ValueError(
+                f"No RulebookProcess found with ID {process_parent_id}"
+            ) from None
+        except models.RulebookProcessQueue.DoesNotExist:
+            raise ValueError(
+                "No Queue associated with RulebookProcess ID "
+                f"{process_parent_id}",
+            ) from None
+        if not hasattr(process, "rulebookprocessqueue"):
+            return None
+        return process.rulebookprocessqueue.queue_name
 
 
+# Internal start/restart requests are sent by the manager in restart_helper.py
 def start_rulebook_process(
-    process_parent_type: ProcessParentType, id: int
+    process_parent_type: ProcessParentType,
+    process_parent_id: int,
 ) -> None:
     """Create a request to start the activation with the given id."""
-    _make_user_request(process_parent_type, id, ActivationRequest.START)
+    requests_queue.push(
+        process_parent_type,
+        process_parent_id,
+        ActivationRequest.START,
+    )
+    RequestDispatcher.dispatch(
```

Comment:
Why is the dispatcher needed?  The task is already pushed to the requests_queue?   Can the workers monitor the requests queue and filter it for their own tasks?

Reply:
In general a node would know if a task is for them.   The same information is already used in the dispatcher, so why not put it in the requests_queue?

Reply:
The only case where we don't know which node to run a task on is a new activation and that can go into an all node queue.

Reply:
The requests queue is only for user requests, we still have to run a management task for monitor running activation which is not an user request. The dispatcher is in charge to know what is the right queue as well as checks if the queue is healthy. 

This implementation attempts to extend the current design and make as few changes as possible. There is a pool of queues and the dispatcher is in charge of selecting the queue in all cases: when a new activation is created, when it is updated and when a user request is executed. 

The queue itself cannot use a balancing system that takes into account the load of the node (the activations that are currently running) as well as being able to discard queues that are considered unhealthy. 

I think it is important to understand the difference between a task to manage an activation and the management of the activation itself. 

eda-server does not manage jobs, jobs are managed by RQ redis. Eda-server manages activations and does it through RQ jobs. RQ queue distribute jobs across their workers. Eda-server distributes activations across queues.